testdata/composition/expected.json

```diff
@@ -151,6 +171,22 @@
                 }
             }
         },
+        "api.FooBarMap": {
+            "type": "object",
+            "properties": {
+                "field3": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "object",
+                        "properties": {
+                            "field4": {
+                                "type": "string"
+                            }
+                        }
+                    }
+                }
+            }
+        },
```

Comment:
Definition 
```
type FooBarMap struct {
	Field3 map[string]MapValue
}

type MapValue struct {
	Field4 string
}
```
should produce 
```  
api.FooBarMap:
    type: object
    properties:
      field3:
        type: object
        additionalProperties:
          $ref: '#/definitions/api.MapValue'

  api.MapValue:
    type: object
    properties:
      field4:
        type: string

```

Reply:
This will register your api.MapValue to definitions
```
		itemSchema, err := parser.parseTypeExpr(pkgName, "", astTypeMap.Value)
		if err != nil {
			return properties, nil, err
		}
```

Neighter of 2 solutions.

You need to use the astTypeMap.Value to add it to object..
```
mapValueScheme :=  &spec.Schema{
	SchemaProps: spec.SchemaProps{
		Ref: spec.Ref{
			Ref: jsonreference.MustCreateRef("#/definitions/" + pkgName + "." + astTypeMap.Value),
		},
	},
}
```

I'm quite busy these days and I don't have much time to spend here. I hope i was clear enough with the last response.

Reply:
You made it clear. Almost did it yourself ><